lib/auth/auth.go

```diff
@@ -4378,6 +4378,47 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 	return req, nil
 }
 
+// getAccessRequestPromotions will return potential access list promotions for an access request.
+func (a *Server) getAccessRequestPromotions(ctx context.Context, req types.AccessRequest) (types.AccessRequest, *types.AccessRequestAllowedPromotions) {
+	reqCopy := req.Copy()
+	promotions, err := modules.GetModules().GenerateAccessRequestPromotions(ctx, a.Services, reqCopy)
+	if err != nil {
+		// Do not fail the request if the promotions failed to generate.
+		// The request promotion will be blocked, but the request can still be approved.
+		log.WithError(err).Warn("Failed to generate access list promotions.")
```

Comment:
IMO, this error should be returned to the caller or at least the function comment should include that errors are being logged and not returned.

Reply:
I'll let the function comment handle this -- I did this to reduce duplication in the create function above.